assets/js/components/notifications/AdSenseAlerts.js

```diff
@@ -39,60 +39,56 @@ function AdSenseAlerts() {
 	const accountID = useSelect( ( select ) =>
 		select( MODULES_ADSENSE ).getAccountID()
 	);
+
+	const [ registeredNotifications, setRegisteredNotifications ] = useState(
+		[]
+	);
+	const { registerNotification } = useDispatch( CORE_NOTIFICATIONS );
+
 	const notifications = useSelect( ( select ) =>
 		select( MODULES_ADSENSE ).getNotifications()
 	);
 
-	if (
-		! adSenseModuleConnected ||
-		! accountID ||
-		notifications === undefined
-	) {
-		return null;
-	}
+	useEffect( () => {
+		// If AdSense is not connected, or the account ID is not set,
+		// there's nothing to do here.
+		if ( ! adSenseModuleConnected || ! accountID ) {
+			return;
+		}
 
-	return (
-		<Fragment>
-			{ notifications.map(
-				( {
-					id,
-					title,
-					description,
-					format,
-					ctaURL,
-					ctaLabel,
-					ctaTarget,
-					severity,
-					isDismissable,
-				} ) => (
-					<BannerNotification
-						className="googlesitekit-adsense-alert"
-						key={ id }
-						id={ id }
-						title={ title || '' }
-						description={ description }
-						WinImageSVG={ NotificationAlertSVG }
-						format={ format || 'small' }
-						ctaLink={ ctaURL }
-						ctaLabel={ ctaLabel }
-						ctaTarget={ ctaTarget }
-						type={ severity }
-						dismiss={ __( 'OK, Got it!', 'google-site-kit' ) }
-						isDismissible={ isDismissable || true }
-						module="adsense"
-						moduleName={ _x(
-							'AdSense',
-							'Service name',
-							'google-site-kit'
-						) }
-						dismissExpires={ 0 }
-						showOnce={ false }
-						logo
-					/>
-				)
-			) }
-		</Fragment>
-	);
+		// Register any notifications from the server that haven't yet been
+		// registered.
+		//
+		// (Usually there will be one, if any, notification from the server.)
+		notifications?.forEach( ( notification ) => {
+			if ( registeredNotifications.includes( notification.id ) ) {
+				return;
+			}
```

Comment:
I'm not sure we need to track the registered IDs since these shouldn't have duplicates. If anything, we can probably store a boolean whether notifications were registered at all so that we only loop through and register them once. Also, `registerNotification` will not enqueue a notification which is already in the queue.

Reply:
This approach allows for more flexibility; if the array of notifications ever changes but has a new ID, it will be registered. But prevents needless dispatches if the notification was already registered.

I don't think it's too complicated to keep around, but if you want we can remove it and allow for redundant `dispatch()` calls with a `registerNotification` action; it just seems a bit better to avoid them but keep this flexibly coded.